src/coreclr/classlibnative/bcltype/arraynative.cpp

```diff
@@ -28,117 +28,17 @@ FCIMPL1(INT32, ArrayNative::GetCorElementTypeOfElementType, ArrayBase* arrayUNSA
 }
 FCIMPLEND
 
-// array is GC protected by caller
-void ArrayInitializeWorker(ARRAYBASEREF * arrayRef,
-                           MethodTable* pArrayMT,
-                           MethodTable* pElemMT)
-{
-    STATIC_CONTRACT_MODE_COOPERATIVE;
-
-    // Ensure that the array element type is fully loaded before executing its code
-    pElemMT->EnsureInstanceActive();
```

Comment:
This call to `EnsureInstanceActive` is important. For example, it is responsible for triggering module initializer if the element constructor is the first code running in the module.